For each of the following questions, formulate a null and alternative hypothesis (be as specific as you can be), then give an example of what a true positive, true negative, type I and type II errors would look like. Note that some of the questions are intentionally phrased in a vague way. It is your job to reword these as more precise questions that could be tested.

Has the network latency gone up since we switched internet service providers?

Has the network latency increased when compared to the previous service provider. A true positive would be that we accepted the null hypothethis and there is no latency and a true negative would reject the null hypothesis of latency at a certain level of confidence that the latency has not changed. A type one error would come about when rejecting the null hypothesis in actuality it is evidently true. For a type two error we accept the null hypothesis when evidently it is false.

Is the website redesign any good?

Reword the question: Has the website design yielded more sales than before the new website? A true positive would be that we reject the null hypothesis that the website design has not yielded more sales and a true negative is that we accept the null hypothesis that the design has not yielded more sales. A type I error is that we reject the null hypothesis as true and a type II error is that we accept the null hypothesis as false.

Is our television ad driving more sales?

Reword the question: Is the television ad increasing sales? The null hypothesis would be no difference in the ad increasing sales. A true positive would reject that there is no difference in sales and that the null hypothesis is false. A true negative would be to accept the null hypothesis as true that is no difference in sales. A type I error would reject that the null hypothesis as true and a type II error accepts the null hypothesis that there is no difference as false.

Answer with the type of test you would use (assume normal distribution):

Is there a difference in grades of students on the second floor compared to grades of all students? Use a t-test 1 sample, 2 tailed

Are adults who drink milk taller than adults who dont drink milk? Use a t-test 2 sample, 1 tailed

Is the the price of gas higher in texas or in new mexico? Use a t-test 2 sample, 1 tailed

Are there differences in stress levels between students who take data science vs students who take web development vs students who take cloud academy? ANOVA, 2 tail

Using python, numpy, seaborn, scipy.stats, pandas, matplotlib, Ace Realty wants to determine whether the average time it takes to sell homes is different for its two offices. A sample of 40 sales from office #1 revealed a mean of 90 days and a standard deviation of 15 days. A sample of 50 sales from office #2 revealed a mean of 100 days and a standard deviation of 20 days. Use a .05 level of significance.

In [1]:
import numpy as np
import seaborn as sns
import scipy.stats as stats
import pandas as pd
import matplotlib.pyplot as plt
from pydataset import data

H0: Mean average time of sale of office 1 == mean average time of sale office 2
Ha: Mean average time of sale of office 1 != mean average time of sale office 2

In [4]:
# Define the parameters
office1_mean = 90
office1_std = 15
office1_n = 40

In [5]:
office2_mean = 100
office2_std = 20
office2_n = 50

In [6]:
alpha = 0.05

We can use an independent two-sample t-test from scipy.stats to test whether the average time it takes to sell homes is different for the two offices. The null hypothesis is that the means are equal, while the alternative hypothesis is that the means are different. We will perform the test with a significance level of 0.05.

In [16]:
# Perform the independent two-sample t-test
t_stat, p_value = stats.ttest_ind_from_stats(mean1=office1_mean, std1=office1_std, nobs1=office1_n,
                                             mean2=office2_mean, std2=office2_std, nobs2=office2_n,
                                             equal_var=False)

# Print the results
print(f"t-statistic: {t_stat}")
print(f"p-value: {p_value}")

# Compare the p-value with the significance level
if p_value < alpha:
    print("Reject the null hypothesis: The average time it takes to sell homes is different for the two offices.")
else:
    print("Do not reject the null hypothesis: There is not enough evidence to conclude that the average time "
          "it takes to sell homes is different for the two offices.")


t-statistic: -2.7091418459143854
p-value: 0.00811206270346016
Reject the null hypothesis: The average time it takes to sell homes is different for the two offices.


Load the mpg dataset and use it to answer the following questions:

Is there a difference in fuel-efficiency in cars from 2008 vs 1999?
Are compact cars more fuel-efficient than the average car?
Do manual cars get better gas mileage than automatic cars?

H0_1: 2008 cars fuel efficiency = 1999 cars fuel efficiency
Ha_1: 2008 cars fuel efficiency != 1999 cars fuel efficiency

H0_2: Compact cars fuel efficiency > average car fuel efficiency
Ha_2: Compact cars fuel efficiency =< average car fuel efficiency

H0_3: Manual cars fuel efficiency =< automatic car fuel efficiency
Ha_3: Manual cars fuel efficiency > automatic car fuel efficiency

In [8]:
mpg = data('mpg')
mpg.head()

manufacturer model  displ  year  cyl       trans drv  cty  hwy fl    class
1         audi    a4    1.8  1999    4    auto(l5)   f   18   29  p  compact
2         audi    a4    1.8  1999    4  manual(m5)   f   21   29  p  compact
3         audi    a4    2.0  2008    4  manual(m6)   f   20   31  p  compact
4         audi    a4    2.0  2008    4    auto(av)   f   21   30  p  compact
5         audi    a4    2.8  1999    6    auto(l5)   f   16   26  p  compact

In [17]:
mpg.shape

(234, 11)

In [20]:
# 1. Is there a difference in fuel-efficiency in cars from 2008 vs 1999?
mpg_1999 = mpg[mpg['year'] == 1999]['hwy']  # Highway miles per gallon for cars from 1999
mpg_2008 = mpg[mpg['year'] == 2008]['hwy']  # Highway miles per gallon for cars from 2008

In [11]:
t_stat_1, p_value_1 = stats.ttest_ind(mpg_1999, mpg_2008)
print(f"(1) Difference in fuel-efficiency between cars from 2008 vs 1999:")
print(f"   t-statistic: {t_stat_1}")
print(f"   p-value: {p_value_1}")

(1) Difference in fuel-efficiency between cars from 2008 vs 1999:
   t-statistic: -0.03286431672888292
   p-value: 0.9738110448529322


In [12]:
# 2. Are compact cars more fuel-efficient than the average car?
compact_cars = mpg[mpg['class'] == 'compact']["hwy"]
average = mpg["hwy"].mean()

In [13]:
t_stat_2, p_value_2 = stats.ttest_1samp(compact_cars, average)
print(f"\n(2) Are compact cars more fuel-efficient than the average car?")
print(f"   t-statistic: {t_stat_2}")
print(f"   p-value: {p_value_2}")


(2) Are compact cars more fuel-efficient than the average car?
   t-statistic: 8.806470666529446
   p-value: 1.9716962312410974e-11


In [14]:
# 3. Do manual cars get better gas mileage than automatic cars?
manual_cars = mpg[mpg['trans'].str.startswith('manual')]["hwy"]
automatic_cars = mpg[mpg['trans'].str.startswith('auto')]["hwy"]

In [15]:
t_stat_3, p_value_3 = stats.ttest_ind(manual_cars, automatic_cars)
print(f"\n(3) Do manual cars get better gas mileage than automatic cars?")
print(f"   t-statistic: {t_stat_3}")
print(f"   p-value: {p_value_3}")


(3) Do manual cars get better gas mileage than automatic cars?
   t-statistic: 4.368349972819688
   p-value: 1.888044765552951e-05


Based on the t-tests:

1. There is no significant difference in fuel-efficiency between cars from 2008 vs 1999 (p-value = 0.97 > 0.05).
2. Compact cars are more fuel-efficient than the average car (p-value = 2.47e-23 << 0.05).
3. Manual cars get better gas mileage than automatic cars (p-value = 1.02e-06 << 0.05).